### PR TITLE
Gauge: Add overflow scrolling support for vertical and horizontal orientations

### PR DESCRIPTION
**Pull request by nmarrs**
Add support to gauge panel to allow for overflow scrolling support. This avoids the visualization breaking / glitching out while in either the vertical or horizontal orientation.

TODO:

- [x] Figure out vertical orientation scaling (text / label aren't scaling correctly)
- [x] Figure out if horizontal / vertical options are switched (create follow-up issue for this if this is the case 🙃)
- [x] Add docs (separate PR probably would be cleanest)

Before
before

After

https://github.com/grafana/grafana/assets/22381771/3b42b4f4-e034-4a83-b9fe-bffbce6d1cd5

Related to https://github.com/grafana/grafana/pull/46819
Fixes https://github.com/grafana/support-escalations/issues/6713

[example dashboard json.txt](https://github.com/grafana/grafana/files/12059475/En.Gauge-1689378489846.txt)

### CODE REVIEW

File: packages/grafana-ui/src/components/Gauge/Gauge.tsx [hunk 1]
```diff
@@ -145,11 +147,15 @@ export class Gauge extends PureComponent<Props> {
   }
 
   renderVisualization = () => {
-    const { width, value, height, onClick, text, theme } = this.props;
+    const { width, value, height, onClick, text, theme, orientation } = this.props;
     const autoProps = calculateGaugeAutoProps(width, height, value.title);
+
+    // If the gauge is in vertical layout, we need to set the width of the gauge to the height of the gauge
+    const gaugeWidth = orientation === VizOrientation.Vertical ? `${autoProps.gaugeHeight}px` : '100%';
```
Comment (nmarrs) [inline packages/grafana-ui/src/components/Gauge/Gauge.tsx:154]: Not confident on this - necessary for vertical mode to render somewhat correctly but scaling of value / label is off when resized

Without this change
Screenshot 2023-07-14 at 5 02 38 PM


With change / scaling issues (for vertical setting (laid out horizontally though...))



Uploading scaling.mov…

File: public/app/plugins/panel/gauge/panelcfg.cue [hunk 2]
```diff
@@ -29,6 +29,8 @@ composableKinds: PanelCfg: {
 					common.SingleStatBaseOptions
 					showThresholdLabels:  bool | *false
 					showThresholdMarkers: bool | *true
+					minVizWidth:          uint32 | *250
+					minVizHeight:         uint32 | *200
```
Comment (ryantxu) [inline public/app/plugins/panel/gauge/panelcfg.cue:33]: maybe 100?  250 seems OK if we could set it only for *new* panels...

**Comment by ryantxu** [comment]
this is interesting, but my first reaction was that the data is missing, not that it is hidden behind scroll bars 😬 

The behavior may seem good when things get really out of hand, but given that our scrollbars only show up when you interact, that makes me a bit scared changing this default behavior for everyone.

Perhaps a very small default?  50? 100? or no default and have it be explicit?

**Comment by ryantxu** [comment]
or maybe some overlay saying "showing 5 of 100 gauges" 🤷🏻

**Comment by nmarrs** [comment]
> The behavior may seem good when things get really out of hand, but given that our scrollbars only show up when you interact, that makes me a bit scared changing this default behavior for everyone.

On my screen it seems like the scrollbars are always present if there is overflow 🤔 
Screenshot 2023-08-02 at 6 26 24 PM

File: public/app/plugins/panel/gauge/panelcfg.cue (hunk 2 above)
Comment (nmarrs) [reply public/app/plugins/panel/gauge/panelcfg.cue]: I'm good with that - here is what that looks like in example (set to 100 vs 200 /250)
Screenshot 2023-08-02 at 6 42 13 PM

**Review by nmarrs — COMMENTED** [review]
It seems that at least for me locally that the scroll bars are always present if there is overflow content. Given this I think we should be ok to avoid including a "x of y gauges are currently visible" label 🤷‍♂️ Thoughts @ryantxu?

**Comment by torkelo** [comment]
Om OS X by default scrollars are only shown on hover (in overlay mode). Usually disabled when you plugin an external mouse

**Comment by jasuade** [comment]
Hey there @nmarrs wanted to check with you about the timeline for this feature, in which version will this be included?

**Comment by nmarrs** [comment]
I looked into adding a display for "you are viewing x out of y gauges" and it wasn't super happy with the number of random edge cases that needed to be covered.

In my mind this in its current state is 100% better than the broken render state before. I reduced the default size some - I think totally removing a default size and causing the visualization to be "broken" by default doesn't make a ton of sense either.

I'm good with merging this and if community provides feedback indicating that not know about scrolling being available then we can address it. This is also solvable with improved panel titles (i.e. dashGPT produced a title including the number of gauges in the panel which would clue user that more content may be available)

**Comment by nmarrs** [comment]
Hi @jasuade - we are targeting to have this functionality released in v10.2

File: public/app/plugins/panel/gauge/panelcfg.cue [hunk 3]
```diff
@@ -29,6 +29,8 @@ composableKinds: PanelCfg: {
 					common.SingleStatBaseOptions
 					showThresholdLabels:  bool | *false
 					showThresholdMarkers: bool | *true
+					minVizWidth:          uint32 | *100
+					minVizHeight:         uint32 | *100
```
Comment (ryantxu) [inline public/app/plugins/panel/gauge/panelcfg.cue:33]:
```suggestion
					minVizWidth:          uint32 | *75
					minVizHeight:         uint32 | *75
```

lets drop to 75 so it really needs to be an explicit setting

**Review by ryantxu — APPROVED** [review]
lets change the default to 75, then I think it looks good 🎉

**Pull request merged**